integration_tests/test_solo_machine.py

```diff
@@ -0,0 +1,179 @@
+        os.environ["SOLO_GRPC_ADDRESS"] = f"http://0.0.0.0:{self.grpc_port}"
+        os.environ["SOLO_RPC_ADDRESS"] = f"http://0.0.0.0:{self.rpc_port}"
+        os.environ["SOLO_TRUSTED_HASH"] = self.trusted_hash
+        os.environ["SOLO_TRUSTED_HEIGHT"] = self.trusted_height
+
+    def prepare(self):
+        self.get_chain_info()
+        self.set_env()
+
+    def run_sub_cmd(self, sub_cmd, json_output=True):
+        if json_output:
+            cmd = f"{self.bin_file} --output json {sub_cmd}"
+        else:
+            cmd = f"{self.bin_file} {sub_cmd}"
+        output = interact(cmd)
+        data = parse_output(output, json_output)
+        return data
+
+    def init(self):
+        sub_cmd = "init"
+        return self.run_sub_cmd(sub_cmd)
+
+    def add_chain(self):
+        sub_cmd = "chain add"
+        return self.run_sub_cmd(sub_cmd)
+
+    def connect_chain(self):
+        sub_cmd = f"ibc connect {self.chain_id}"
+        result = self.run_sub_cmd(sub_cmd, False)
+        if "error" in result:
+            raise Exception(result)
+
+    def get_balance(self):
+        sub_cmd = f"chain balance {self.chain_id} solotoken"
+        return self.run_sub_cmd(sub_cmd)["data"]["balance"]
+
+    # fixme after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def mint(self, amount="0x20"):
+        sub_cmd = f"ibc mint {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+    # fixme fix after this PR merged, currently only support hex string amount
+    # https://github.com/crypto-com/ibc-solo-machine/pull/82
+    def burn(self, amount="0x10"):
+        sub_cmd = f"ibc burn {self.chain_id} {amount} solotoken"
+        return self.run_sub_cmd(sub_cmd)
+
+
+def test_solo_machine(cluster, tmpdir_factory):
+    """
+    check solo machine
+    """
+    cli = cluster
+    wait_for_block(cli, 1)
+
+    # get the chain balance
+    chain_solo_addr = cluster.address("solo-signer")
+    chain_balance_1 = cluster.balance(chain_solo_addr)
+    assert chain_balance_1 == 1500 * 10 ** 8
+
+    base_port = cli.base_port(0)
+    wait_for_port(ports.grpc_port(base_port))
+    tmp_path = tmpdir_factory.mktemp("db")
+    solo_machine = SoloMachine(tmp_path, base_port=base_port)
+    solo_machine.prepare()
+    data = solo_machine.init()
+    assert data["result"] == "success"
+    data = solo_machine.add_chain()
+    assert data["result"] == "success"
+    wait_for_block(cli, 2)
+    solo_machine.connect_chain()
+    wait_for_block(cli, 2)
+    balance_0 = int(solo_machine.get_balance())
+    assert balance_0 == 0
+
+    # mint
+    data = solo_machine.mint(20)
+    assert data["result"] == "success"
+    assert int(data["data"]["amount"], 16) == 20
+    wait_for_block(cli, 2)
+    balance_1 = int(solo_machine.get_balance())
+    assert balance_1 == 20
+
+    # check the chain balance
+    chain_balance_2 = cli.balance(chain_solo_addr)
```

Comment:
besides checking the on-chain balance, it'll be good to assert the denomination is correct as well as well